app/pb_kits/playbook/pb_button/_button.jsx

```diff
@@ -27,62 +29,64 @@ type ButtonPropTypes = {
 
 const buttonClassName = (props: ButtonPropTypes) => {
   const {
-    dark=false,
-    disabled=false,
-    fullWidth=false,
-    loading=false,
-    size=null,
-    type='inline',
-    variant='primary',
+    dark = false,
+    disabled = false,
+    fullWidth = false,
+    loading = false,
+    size = null,
+    type = 'inline',
+    variant = 'primary',
   } = props
 
   let className = 'pb_button_kit'
 
-  className    += `${variant !== null ? `_${variant}` : ''}`
-  className    += `${type !== null ? `_${type}` : ''}`
-  className    += `${size !== null ? `_${size}` : ''}`
-  className    += `${dark === true ? '_dark' : ''}`
-  className    += `${fullWidth ? '_block' : ''}`
-  className    += disabled ? '_disabled' : '_enabled'
-  className    += loading ? '_loading' : ''
+  className += `${variant !== null ? `_${variant}` : ''}`
+  className += `${type !== null ? `_${type}` : ''}`
+  className += `${size !== null ? `_${size}` : ''}`
+  className += `${dark === true ? '_dark' : ''}`
+  className += `${fullWidth ? '_block' : ''}`
+  className += disabled ? '_disabled' : '_enabled'
+  className += loading ? '_loading' : ''
 
   return className
 }
```

Comment:
```javascript
import { compact } from 'lodash'

const buttonClassName = ({
  dark = true,
  disabled = false,
  fullWidth = false,
  loading = false,
  size = null,
  type = 'inline',
  variant = 'primary',
}) => (
  compact([
    'pb_button_kit', variant, type, size,
    disabled ? 'disabled' : 'enabled',
    dark && 'dark',
    fullWidth && 'block',
    loading && 'loading',
  ]).join('_')
)
```

Reply:
I was not planning to refactor the existing implementation to get the `onClick` change out soon. However, if this component isn't widely used in Nitro yet then I'll take the refactoring path.

Reply:
I am on the fence tbh. I did a quick search:

```
> ag -Q "pb_rails(\"button"

components/nitro_component_transition/app/views/bootstrap/survey_campaigns/index.html.erb
9:<%= pb_rails("button", props: { text: "View Survey Invitation Queue", link: queue_survey_campaigns_path}) %>

components/nitro_component_transition/app/views/bootstrap/installation_crews/_date_range.html.erb
42:        <%= pb_rails("button", props: { variant: "secondary", full_width: true }) do %>

components/contact_center/app/views/contact_center/dispositions/index.html.erb
5:        <%= pb_rails("button", props: { variant: "secondary", link: contact_center.new_disposition_url }) do %>
34:            <%= pb_rails("button", props: { variant: "secondary", link: contact_center.edit_disposition_url(disposition) }) do %>

components/admin/app/views/admin/app_settings/_goals_form.html.erb
100:      <%= pb_rails("button", props: { text: "Save Changes" }) %>
```

Reply:
Oh, I was actually referring to the React version only. I was doing some digging in `nitro-web` and there seems to be no references to the Playbook React button.

I'm considering refactoring the React button applying a few opinionated ideas and have it out there for a quick discussion with the rest of the devs.

Let me know if anyone objects to that idea.